### DISCUSSION TITLE
MapCompose 2

### DISCUSSION THREAD

**Discussion by Nohus** [peterLaurence/MapCompose#32] [General]
The first issue is that creating a map and not adding any layer results in a crash:
```
java.lang.IllegalStateException: Compose Runtime internal error. Unexpected or incorrect use of the Compose internal runtime API (Start/end imbalance). Please report to Google or use https://goo.gle/compose-feedback
```
The stacktrace contains no app code and no library code, it's a crash of Compose. The library shouldn't crash when there are no layers, just display nothing.

**Comment by p-lr** [comment]
I've fixed it, along with the gesture break in my last two commits on `v2`.

**Reply by Nohus** [reply]
Thanks, can confirm it's fixed.

**Comment by p-lr** [comment]
As you can see, I'm currently fixing issues in `v2` branch only. However, I'll backport some fixes to `master` right before publishing the last version of 1.x.x branch.

**Comment by p-lr** [comment]
I'm currently facing another issue which is kind of a blocker for me. `v2` is based on Compose 1.1.0 beta04, and we're sometimes hitting this [bug](https://issuetracker.google.com/issues/208855077) using the markers API. I may have to revert to using Compose stable for `v2`.
To be able to continue developments in my app, I had to manually downgrade `v2` to use Compose stable and Kotlin 1.5.31 (also revert the `TileStreamProvider` to non suspend). I haven't committed anything, hoping that will be fixed in 1.1.0-beta05.

**Reply by Nohus** [reply]
Looks like the bug you refer to was fixed yesterday, maybe it's worth just waiting for the next release of Compose with the release of `v2`?

**Reply by p-lr** [reply]
I hope so, yes.

**Comment by p-lr** [comment]
If everything goes well after you test `init-values`, I'll release `2.0.0-beta02` using `Compose 1.1.0-rc1` which was released yesterday.

**Reply by Nohus** [reply]
I see it got released, great!

**Comment by p-lr** [comment]
As you can see, I've changed my user name for `p-lr`. This is something I wanted to do for a while. You might have to update your local git remote (in the unlikely but possible event that someone picks my old username).
